src/flatMap.test.ts

```diff
@@ -3,6 +3,14 @@ import { find } from './find';
 import { flatMap } from './flatMap';
 import { pipe } from './pipe';
 
+it('should accept fn returning a readonly array', () => {
+  const result = flatMap(
+    [1, 2] as const,
+    x => [x * 2, x * 3] as ReadonlyArray<number>
```

Comment:
would be enough to just do as const here
```suggestion
    x => [x * 2, x * 3] as const
```